### PR TITLE
[Maistra-1264] Update 1.4.6 to revert a custom value that we use for a test in Maistra to the upstream value

### PR DESCRIPTION
**Pull request by brian-avery**
**What this PR does / why we need it**:
This fixes a test to match upstream rather than what we had in Maistra.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: src/envoy/tcp/sni_verifier/sni_verifier_test.cc [hunk 1]
```diff
@@ -136,7 +136,7 @@ INSTANTIATE_TEST_SUITE_P(
                     std::make_tuple(TLS1_2_VERSION, TLS1_2_VERSION),
                     std::make_tuple(TLS1_3_VERSION, TLS1_3_VERSION)));
 
-TEST_P(SniVerifierFilterTest, SnisMatch) {
+TEST_P(SniVerifierFilterTest, DISABLED_SnisMatch) {
```
Comment (jwendell) [inline src/envoy/tcp/sni_verifier/sni_verifier_test.cc:139]: Is there an way we can keep track of those disabled tests and not forget about them?
Perhaps by adding a comment to all disabled tests in this file like `FIXME: Tests temporarily disabled, see: https://issues.redhat.com/projects/MAISTRA/issues/MAISTRA-XXXX` ? (Then we would need to create this issue and link it here)

Comment (brian-avery) [reply src/envoy/tcp/sni_verifier/sni_verifier_test.cc:139]: Done. Added a comment.

Comment (jwendell) [reply src/envoy/tcp/sni_verifier/sni_verifier_test.cc:139]: Thanks!

**Comment by brian-avery** [comment]
As there are 10 disabled tests, and only DISABLED_SniTooLarge has two failures, backing out the change from TOO_LARGE_SERVER_NAME_SIZE to TLS_MAX_CLIENT_HELLO while we address the 10 failures.

**Comment by brian-avery** [comment]
@mergifyio refresh

**Pull request merged**